core/arch/arm/kernel/thread.c

```diff
@@ -1521,8 +1521,9 @@ void thread_get_user_kdata(struct mobj **mobj, size_t *offset,
 
 	core_mmu_get_user_va_range(&v, NULL);
 	*va = v + thread_user_kcode_size;
-	*mobj = mobj_tee_ram;
-	*offset = (vaddr_t)thread_user_kdata_page - VCORE_START_VA;
+	*mobj = mobj_tee_ram_rw;
+	*offset = (vaddr_t)thread_user_kdata_page -
+			(vaddr_t)mobj_get_va(*mobj, 0);
```

Comment:
`thread_user_kdata_page` is an array so that will not work as intended, but we could at least have it aligned  like:
``` C
        *offset = (vaddr_t)thread_user_kdata_page - 
                  (vaddr_t)mobj_get_va(*mobj, 0);
```